From ef4be7b70e0132a3acc487fe02abcbc4abf84501 Mon Sep 17 00:00:00 2001
From: Jouni Malinen <quic_jouni@quicinc.com>
Date: Fri, 24 Jan 2025 18:22:36 +0200
Subject: [PATCH] wlantest: Parse Extended MLD Capabilities And Operations in
 Common Info

Show the value in a clearer debug print instead of the generic "Extra
information at the end of Common Info".

Signed-off-by: Jouni Malinen <quic_jouni@quicinc.com>
---
 wlantest/rx_mgmt.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/wlantest/rx_mgmt.c b/wlantest/rx_mgmt.c
index 0f00a8834..c9d0e1bee 100644
--- a/wlantest/rx_mgmt.c
+++ b/wlantest/rx_mgmt.c
@@ -197,6 +197,18 @@ static void parse_basic_ml(const u8 *ie, size_t len, bool ap,
 		pos++;
 	}
 
+	if (ctrl & BASIC_MULTI_LINK_CTRL_PRES_EXT_MLD_CAP) {
+		if (ci_end - pos < 2) {
+			wpa_printf(MSG_INFO,
+				   "No room for Extended MLD Capabilities And Operations in Multi-Link Common Info");
+			return;
+		}
+		wpa_printf(MSG_DEBUG,
+			   "Extended MLD Capabilities And Operations: 0x%x",
+			   WPA_GET_LE16(pos));
+		pos += 2;
+	}
+
 	if (pos < ci_end) {
 		wpa_hexdump(MSG_INFO,
 			    "Extra information at the end of Common Info",
-- 
2.43.0

